src/bin/rad-ci.rs

```diff
@@ -0,0 +1,161 @@
+use std::net::TcpStream;
+
+use rad_ci::{Options, HELP};
+use rad_terminal::components as term;
+
+use anyhow::Context;
+use ssh2::Session;
+
+fn main() {
+    rad_terminal::args::run_command::<Options, _>(HELP, "CI", run);
+}
+
+fn run(options: Options) -> anyhow::Result<()> {
+    let seed = options.seed.context("Seed is invalid")?;
+    // SSH into server
+    let tcp = TcpStream::connect(format!("{}:22", &seed))?;
+    let mut sess = Session::new()?;
+    sess.set_tcp_stream(tcp);
+    sess.handshake()?;
+
+    // Authenticate
+    let spinner = term::spinner(&format!("SSH into remote {}", &seed));
+    if options.ssh_index == 0 {
+        sess.userauth_agent(&options.ssh_user)?;
+    } else {
+        let mut agent = sess.agent()?;
+        agent.connect()?;
+        agent.list_identities()?;
+        let identities = agent.identities()?;
+        let identity = identities.get(options.ssh_index).context(format!(
+            "Couldn't retrieve ssh-agent key #{}",
+            options.ssh_index
+        ))?;
+        agent.userauth(&options.ssh_user, identity)?;
+    }
+    if sess.authenticated() {
+        spinner.finish();
+    } else {
+        spinner.failed();
+    }
+
+    let verbosity = options.verbose;
+    let execute_fn = |cmd, spinner| execute_cmd_with_spinner(&sess, cmd, spinner, verbosity);
+
+    // Check requirements
+    term::blank();
+    term::info!("Checking requirements:");
+    term::blank();
+
+    let mut requirements = vec!["wget", "docker", "docker-compose"];
+    if options.config.is_some() {
+        requirements.remove(0);
+    }
+
+    if requirements
+        .iter()
+        .map(|req| {
+            let spinner = term::spinner(&format!(
+                "Checking if {} exists",
+                term::format::highlight(req)
+            ));
+            let (_, status) =
+                execute_cmd_with_spinner(&sess, &format!("which {}", req), spinner, verbosity);
+            status
+        })
+        .sum::<i32>()
+        > 0
+    {
+        term::info!("Requirements are not installed.");
+        std::process::exit(1);
+    }
+    term::blank();
+
+    // Upload file if supplied
+    if options.config.is_some() {
+        let path = options.config.unwrap();
+        let mut file = std::fs::File::open(path).context("Couldn't open config file")?;
+        let mut content = String::new();
+        file.read_to_string(&mut content)
+            .context("Couldn't read the contents of config file")?;
+
+        let mut remote_file = sess
+            .scp_send(
+                std::path::Path::new("/etc/concourse-docker-compose.yml"),
+                0o644,
+                content.len().try_into().unwrap(),
+                None,
+            )
+            .unwrap();
+
+        remote_file.write_all(content.as_bytes()).unwrap();
+
+        remote_file.send_eof().unwrap();
+        remote_file.wait_eof().unwrap();
+        remote_file.close().unwrap();
+        remote_file.wait_close().unwrap();
```

Comment:
You can get rid of the unwraps here and in other places.

Reply:
like this? https://github.com/radicle-dev/radicle-client-tools/pull/35/commits/21aa4f338dfbf8b7570287a4f7701d4462f48278#diff-2f5300f7dedf306f4b73a94a6c5c3674f3ece69ed4bda39a3b27b8448149d978R201-R204

Reply:
Heh no, by using the try operator: `remote_file.send_eof()?` etc.